File: python/wxwidgets/counter/main_frame.py
import wx


class MainFrame(wx.Frame):
    counter: int = 0

    def __init__(self, *args, **kw):
        super(MainFrame, self).__init__(*args, **kw)

        panel = wx.Panel(self)

        self.input = wx.TextCtrl(panel, value=str(self.counter))
        self.input.Enable(False)

        btn = wx.Button(panel, label="Counter")

        v_sizer = wx.BoxSizer(wx.VERTICAL)
        h_sizer = wx.BoxSizer(wx.HORIZONTAL)

        h_sizer.Add(self.input, 40, wx.ALIGN_CENTER)
        h_sizer.AddStretchSpacer(1)
        h_sizer.Add(btn, 20, wx.ALIGN_CENTER)

        v_sizer.AddStretchSpacer()
        v_sizer.Add(h_sizer, 0, wx.ALIGN_CENTER)
        v_sizer.AddStretchSpacer()

        self.Bind(wx.EVT_BUTTON, self.on_click)

        panel.SetSizer(v_sizer)

    def on_click(self, _evt):
        self.counter += 1
        self.input.SetValue(str(self.counter))

File: cpp/wxwidgets/counter/MainFrame.cpp
#include "MainFrame.h"

// Custom IDs must be > 1 and < 4999 (wxID_LOWEST)
enum IDs {
    ID_BTN = 2
};

wxBEGIN_EVENT_TABLE(MainFrame, wxFrame)
                EVT_BUTTON(ID_BTN, MainFrame::onBtnClick)
wxEND_EVENT_TABLE()

MainFrame::MainFrame(const wxString &title) : wxFrame(nullptr, wxID_ANY, title) {
    auto *pPanel = new wxPanel(this);

    const wxString &initVal = std::to_string(counter);
    txtCtrl = new wxTextCtrl(pPanel, wxID_ANY, initVal);
    txtCtrl->Enable(false);

    auto pButton = new wxButton(pPanel, ID_BTN, "Counter");

    auto *vSizer = new wxBoxSizer(wxVERTICAL);
    auto *hSizer = new wxBoxSizer(wxHORIZONTAL);

    hSizer->Add(txtCtrl, 2, wxALIGN_CENTER);
    hSizer->AddSpacer(5);
    hSizer->Add(pButton, 1, wxALIGN_CENTER);

    vSizer->AddStretchSpacer();
    vSizer->Add(hSizer, 0, wxALIGN_CENTER);
    vSizer->AddStretchSpacer();

    pPanel->SetSizer(vSizer);
}

void MainFrame::onBtnClick(wxCommandEvent &) {
    counter++;
    txtCtrl->SetValue(std::to_string(counter));
}
